src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -41,6 +45,10 @@ public async Task Invoke(HttpContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            // Flag to indicate to other systems, e.g. MVC, that authorization middleware was run for this request
+            context.Items[AuthorizationMiddlewareInvokedKey] = AuthorizationMiddlewareInvokedValue;
```

Comment:
Forgive me if it's a stupid question, but what's the point of adding an empty object as a value if we are only checking ContainsKey? Can't we set the value to null and get rid of the readonly property?

Reply:
This code is checking with ContainsKey, but someone else might do something like:

```cs
if (context.Items["__AuthorizationMiddlewareInvoked"] != null)
{
    return;
}
```

And I'm setting a stored an object rather than doing something like `context.Items["key"] = true` because putting true in a `Dictionary<object, object>` would allocate when it boxes the boolean.

Reply:
To fix the wrong check by others it should be enough to expose an extension method but anyway it's not a big deal!